presto-main/src/main/java/com/facebook/presto/server/QuerySessionSupplier.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.metadata.SessionPropertyManager;
+import com.facebook.presto.security.AccessControl;
+import com.facebook.presto.spi.QueryId;
+import com.facebook.presto.spi.security.Identity;
+import com.facebook.presto.transaction.TransactionManager;
+
+import javax.inject.Inject;
+
+import java.util.Locale;
+import java.util.Map;
+
+import static com.facebook.presto.spi.type.TimeZoneKey.getTimeZoneKey;
+import static java.util.Objects.requireNonNull;
+
+public class QuerySessionSupplier
+        implements SessionSupplier
+{
+    private final TransactionManager transactionManager;
+    private final AccessControl accessControl;
+    private final SessionPropertyManager sessionPropertyManager;
+
+    @Inject
+    public QuerySessionSupplier(
+            TransactionManager transactionManager,
+            AccessControl accessControl,
+            SessionPropertyManager sessionPropertyManager)
+    {
+        this.transactionManager = requireNonNull(transactionManager, "transactionManager is null");
+        this.accessControl = requireNonNull(accessControl, "accessControl is null");
+        this.sessionPropertyManager = requireNonNull(sessionPropertyManager, "sessionPropertyManager is null");
+    }
+
+    @Override
+    public Session createSession(
+            QueryId queryId,
+            SessionContext context)
+    {
+        Identity identity = context.getIdentity();
+        accessControl.checkCanSetUser(identity.getPrincipal().orElse(null), identity.getUser());
+
+        Session.SessionBuilder sessionBuilder = Session.builder(sessionPropertyManager)
```

Comment:
Static import `SessionBuilder`

Reply:
We need to import Session anyways. Is that really necessary?